### PR TITLE
Account service without juggler

### PR DESCRIPTION
**Pull request by b-admike**
connect to https://github.com/strongloop-internal/scrum-apex/issues/215
connect to https://github.com/strongloop-internal/scrum-apex/issues/228

### CODE REVIEW

File: services/accountsanjuggler/repositories/account/models/account/model-definition.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "avgBalance": {
+      "type": "number",
+      "required": true
+    },
+    "minimumBalance": {
+      "type": "number",
+      "required": true
+    }
+  }
+}
```
Comment (kjdelisle) [inline services/accountsanjuggler/repositories/account/models/account/model-definition.json]: missing newline

File: services/accountsansjuggler/controllers/AccountController.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    return await this.repository.create(accountInstance);
+  }
+
+  async updateAccount(id, data) {
+    return await this.repository.updateById(id, data);
+  }
+
+   async deleteAccount(id) {
+    return await this.repository.deleteById(id);
+}
```
Comment (raymondfeng) [inline services/accountsansjuggler/controllers/AccountController.ts]: Please fix the indentation.

File: services/accountsansjuggler/bin/get-account [hunk 3]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
```
Comment (raymondfeng) [inline services/accountsansjuggler/bin/get-account]: A better name for `accountsanjuggler`?

File: services/accountsansjuggler/repositories/account/datasources/mysqlds.ts [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+    Class, 
+    DataSource, 
+    Entity, 
+    Filter, 
+    ObjectType, 
+    Options, 
+    Where
+} from 'loopback-next/packages/repository';
+
+export class MySqlConn implements CrudConnector {
```
Comment (raymondfeng) [inline services/accountsansjuggler/repositories/account/datasources/mysqlds.ts]: Please refactor it into a new file.

File: services/accountsansjuggler/repositories/account/datasources/mysqlds.ts [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+      ping(){
+          let self = this;
+         return new Promise<void>(function(resolve, reject) {
+              self.connection.ping(function(err) {
+                  if (err !== null) return reject(err);
+                  resolve(err);
+              })
+          })
+      }
+
```
Comment (raymondfeng) [inline services/accountsansjuggler/repositories/account/datasources/mysqlds.ts]: Please format the code with correct indentation. If you use VSCode, try `prettier javascript formatter`.

File: services/account-without-juggler/repositories/account/datasources/mysql.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
+    "host":"localhost",
+    "user":"root",
+    "password": "dis-mysql",
+    "database": "test-db",
+    "port": 3306
+}
```
Comment (ssh24) [inline services/account-without-juggler/repositories/account/datasources/mysql.json]: Need a new line here.

File: services/account-without-juggler/repositories/account/datasources/mysqlconn.ts [hunk 7]
```diff
@@ -0,0 +1,229 @@
+const debug = require("debug")("accountsansjuggler");
```
Comment (ssh24) [inline services/account-without-juggler/repositories/account/datasources/mysqlconn.ts]: Should we change the debug tag to `account-without-juggler` instead?

File: services/account-without-juggler/repositories/account/models/account/model-definition.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    "avgBalance": {
+      "type": "number",
+      "required": true
+    },
+    "minimumBalance": {
+      "type": "number",
+      "required": true
+    }
+  }
+};
```
Comment (ssh24) [inline services/account-without-juggler/repositories/account/models/account/model-definition.js]: Need new line here.

File: services/account-without-juggler/test/mocha.opts [hunk 9]
```diff
@@ -0,0 +1,2 @@
+--compilers ts:ts-node/register
+--recursive
```
Comment (ssh24) [inline services/account-without-juggler/test/mocha.opts]: Ditto.

File: services/account-without-juggler/test/unit/account-controller.ts [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+  it("updates an account instance", async () => {
+    const result = await accCtrl.updateAccount('{"id":"test1"}', {
+      balance: 2000
+    });
+    expect(result.count).to.be.equal(1);
+    const getResult = await accCtrl.getAccount('{"where":{"id":"test1"}}');
+    expect(getResult).to.not.be.empty();
+    expect(getResult).have.lengthOf(1);
+    expect(getResult[0].id).to.be.equal(testAcc.id);
+    expect(JSON.parse(JSON.stringify(getResult[0])).balance).to.be.equal(2000);
```
Comment (ssh24) [inline services/account-without-juggler/test/unit/account-controller.ts]: `getResult[0].toObject().balance` should work.

File: services/account-without-juggler/tsconfig.json [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+    ],
+    "inlineSources": true
+  },
+  "include": [
+    "**/*"
+  ],
+  "exclude": [
+    "**/*.d.ts"
+  ]
+}
```
Comment (ssh24) [inline services/account-without-juggler/tsconfig.json]: Ditto.

File: services/account-without-juggler/tslint.full.json [hunk 12]
```diff
@@ -0,0 +1,20 @@
+{
+  "$schema": "http://json.schemastore.org/tslint",
+  "extends": [
+    "./tslint.json"
+  ],
+  // This configuration files enabled rules which require type checking
```
Comment (ssh24) [inline services/account-without-juggler/tslint.full.json]: Are these comments needed?

File: services/account-without-juggler/tslint.json [hunk 13]
```diff
@@ -0,0 +1,72 @@
+{
+  // See https://palantir.github.io/tslint/rules/
```
Comment (ssh24) [inline services/account-without-juggler/tslint.json]: Ditto.

File: services/account-without-juggler/index.ts [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+import { accountRepository } from './repositories/account';
+
+class AccountMicroservice extends Application {
+  private _startTime: Date;
+
+  constructor() {
+    super();
+    const app = this;
+    app.controller(AccountController);
+    app.bind('repositories.account').toClass(accountRepository);
```
Comment (bajtos) [inline services/account-without-juggler/index.ts]: It would be nice to introduce `app.repository` API to take care of this.

```ts
app.repository(accountRepository);
```

The argument should be strongly typed to a base Repository interface, which is something that we cannot achieve via `bind().toClass()`, at least not now.

File: services/account-without-juggler/package.json [hunk 15]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "account",
+  "version": "1.0.0",
+  "description": "The Account microservice.",
+  "main": "index.ts",
+  "dependencies": {
+    "@loopback/testlab": "^4.0.0-alpha.5",
+    "@types/mocha": "^2.2.41",
```
Comment (bajtos) [inline services/account-without-juggler/package.json]: I think both `testlab` and `mocha` should go to `devDependencies`, shouldn't it? Why do you actually need `@types/mocha`?

File: services/account-without-juggler/repositories/account/datasources/mysqlconn.ts [hunk 16]
```diff
@@ -0,0 +1,229 @@
...
+        }
+      );
+    }
+  }
+  name: "mysql";
+  interfaces?: string[];
+  connect() {
+    let self = this;
+    return new Promise<void>(function(resolve, reject) {
+      self.connection.connect(function(err: any) {
```
Comment (bajtos) [inline services/account-without-juggler/repositories/account/datasources/mysqlconn.ts]: Have you considered using https://www.npmjs.com/package/mysql-promise or https://www.npmjs.com/package/promise-mysql, instead of this manual way of adding promise support?

File: services/account-without-juggler/repositories/account/datasources/mysqlconn.ts [hunk 17]
```diff
@@ -0,0 +1,229 @@
...
+    let self = this;
+    let sqlStmt = "INSERT INTO " + modelClass.name + " VALUES ";
+    var vals = [];
+    for (var prop in entity) {
+      vals.push(entity[prop]);
+    }
+    sqlStmt += "(" + vals.join(",") + ")" + ";";
+    debug("insert stmt ", sqlStmt);
+    return new Promise<Account>(function(resolve, reject) {
+      self.connection.query(sqlStmt, function(err: any, count: any) {
```
Comment (bajtos) [inline services/account-without-juggler/repositories/account/datasources/mysqlconn.ts]: Have you considered using https://www.npmjs.com/package/@types/mysql to get better type information? (I don't know how it will work with promise-wrapped mysql packages though.)

File: services/account-without-juggler/repositories/account/datasources/mysqlconn.ts [hunk 18]
```diff
@@ -0,0 +1,229 @@
...
+    throw new Error("Not implemented yet.");
+  }
+
+  public create(
+    modelClass: Class<Entity>,
+    entity: EntityData,
+    options: Options
+  ): Promise<EntityData> {
+    let self = this;
+    let sqlStmt = "INSERT INTO " + modelClass.name + " VALUES ";
```
Comment (bajtos) [inline services/account-without-juggler/repositories/account/datasources/mysqlconn.ts]: I think you should escape `modelCass.name` in order to prevent SQL script injection. (Class names probably cannot be changed so much as to cause security problem, but I can imagine there may be valid class names that cannot be used in SQL without escaping.)

File: services/account-without-juggler/repositories/account/datasources/mysqlconn.ts [hunk 19]
```diff
@@ -0,0 +1,229 @@
...
+    var vals = [];
+    for (var prop in entity) {
+      vals.push(entity[prop]);
+    }
+    sqlStmt += "(" + vals.join(",") + ")" + ";";
```
Comment (bajtos) [inline services/account-without-juggler/repositories/account/datasources/mysqlconn.ts]: SQL injection.

File: services/account-without-juggler/repositories/account/datasources/mysqlconn.ts [hunk 20]
```diff
@@ -0,0 +1,229 @@
...
+  ): Promise<EntityData[]> {
+    let self = this;
+    let sqlStmt = "SELECT * FROM " + modelClass.name;
+    if (filter.where) {
+      let sql = "?? = ?";
+      let formattedSql = "";
+      for (var key in filter.where) {
+        formattedSql = mysql.format(sql, [key, filter.where[key]]);
+      }
+      sqlStmt += " WHERE " + formattedSql;
```
Comment (bajtos) [inline services/account-without-juggler/repositories/account/datasources/mysqlconn.ts]: FWIW, I think this is suboptimal performance wise. Each query will be a unique string, therefore the SQL engine has very little opportunities for caching pre-computed values.

Ideally, the SQL statement should looks like this:

```
SELECT * FROM model WHERE foo = ? AND bar = ?
```

The same comment applies to all other places where you are creating custom SQL statements.

File: services/account-without-juggler/repositories/account/datasources/mysqlconn.ts [hunk 21]
```diff
@@ -0,0 +1,229 @@
...
+    modelClass: Class<Entity>,
+    id: Number,
+    data: EntityData,
+    options: Options
+  ): Promise<boolean> {
+    let self = this;
+    let sqlStmt = "UPDATE " + modelClass.name + " SET ";
+    var vals = [];
+    for (var prop in data) {
+      vals.push(prop + "=" + data[prop]);
```
Comment (bajtos) [inline services/account-without-juggler/repositories/account/datasources/mysqlconn.ts]: SQL injection.

File: services/account-without-juggler/repositories/account/datasources/mysqlconn.ts [hunk 22]
```diff
@@ -0,0 +1,229 @@
...
+    }
+    sqlStmt += vals.join(",") + " WHERE " + "`id`=" + id + ";";
```
Comment (bajtos) [inline services/account-without-juggler/repositories/account/datasources/mysqlconn.ts]: SQL injection - pass `id` as a parameter, let the driver to escape it.

File: services/account-without-juggler/repositories/account/datasources/mysqlconn.ts [hunk 23]
```diff
@@ -0,0 +1,229 @@
...
+    throw new Error("Not implemented yet.");
+  }
+
+  public deleteById(
+    modelClass: Class<Entity>,
+    id: Number,
+    options: Options
+  ): Promise<boolean> {
+    let self = this;
+    let sqlStmt = "DELETE FROM " + modelClass.name + " WHERE `id`=" + id + ";";
```
Comment (bajtos) [inline services/account-without-juggler/repositories/account/datasources/mysqlconn.ts]: SQL injection - pass id as a parameter, let the driver to escape it.

File: services/account-without-juggler/test/unit/account-controller.ts [hunk 24]
```diff
@@ -0,0 +1,73 @@
+import { AccountController } from "../../controllers/AccountController";
```
Comment (bajtos) [inline services/account-without-juggler/test/unit/account-controller.ts]: We use single quotes for string delimiters in LoopBack code (and in Node.js code in general). This comment applies to all other JS/TS source files in this pull request too.

File: services/account-without-juggler/tslint.json (hunk 13 above)
Comment (bajtos) [inline services/account-without-juggler/tslint.json]: I think these comments were copied from loopback-next. I added this particular comment in loopback-next to make it easier for people that are not familiar with tslint to find a reference of all rules.

What are your objections against having these comments in place?

Comment (ssh24) [inline services/account-without-juggler/tslint.json]: Ah, I see. Disregard my comments in that case.

File: services/account-without-juggler/index.ts (hunk 14 above)
Comment (b-admike) [inline services/account-without-juggler/index.ts]: @raymondfeng thoughts on the above API to take care of this action?

File: services/account-without-juggler/repositories/account/datasources/mysqlconn.ts [hunk 25]
```diff
@@ -0,0 +1,201 @@
...
+  ObjectType,
+  Options,
+  Where
+} from "loopback-next/packages/repository";
+
+export class MySqlConn implements CrudConnector {
+  private connection: any;
+
+  constructor(config: Object) {
+    db.configure(config, require("mysql2"));
```
Comment (ssh24) [inline services/account-without-juggler/repositories/account/datasources/mysqlconn.ts]: Are we using the `mysql2` driver?

File: services/account-without-juggler/controllers/AccountController.ts [hunk 26]
```diff
@@ -0,0 +1,30 @@
+import { api } from 'loopback-next/packages/core';
+import { def } from './AccountController.api';
+import { accountRepository } from '../repositories/account';
+import { inject } from 'loopback-next/packages/context';
+
+@api(def)
+export class AccountController {
+  @inject('repositories.account') private repository: accountRepository;
```
Comment (superkhau) [inline services/account-without-juggler/controllers/AccountController.ts]: move into constructor inject so that repository can be replaced by passing in

File: services/account-without-juggler/index.ts (hunk 14 above)
Comment (superkhau) [inline services/account-without-juggler/index.ts]: I think we can land and create an issue (feature) for the sugar API mentioned by @bajtos. This PR is huge already.

File: services/account-without-juggler/repositories/account/datasources/mysqlconn.ts (hunk 25 above)
Comment (b-admike) [inline services/account-without-juggler/repositories/account/datasources/mysqlconn.ts]: Yes

File: services/account-without-juggler/repositories/account/datasources/mysql.json [hunk 27]
```diff
@@ -0,0 +1,7 @@
+{
+    "host":"localhost",
+    "user":"root",
+    "password": "dis-mysql",
+    "database": "test-db",
```
Comment (superkhau) [inline services/account-without-juggler/repositories/account/datasources/mysql.json]: let's make it simpler for future users

password: "password" // password
database: "loopback-next-example" // name of example

should be consistent across all examples

File: services/account-without-juggler/repositories/account/datasources/mysqlconn.ts [hunk 28]
```diff
@@ -0,0 +1,201 @@
+const debug = require("debug")("account-without-juggler");
```
Comment (superkhau) [inline services/account-without-juggler/repositories/account/datasources/mysqlconn.ts]: const debug = require('debug')('loopback:repositories:account:datasources:connections:mysql');

Single quote everything in this file too

File: services/account-without-juggler/repositories/account/datasources/mysqlconn.ts [hunk 29]
```diff
@@ -0,0 +1,201 @@
...
+  Entity,
+  EntityData,
+  Filter,
+  ObjectType,
+  Options,
+  Where
+} from "loopback-next/packages/repository";
+
+export class MySqlConn implements CrudConnector {
+  private connection: any;
```
Comment (superkhau) [inline services/account-without-juggler/repositories/account/datasources/mysqlconn.ts]: : Connection, let's get away from any

File: services/account-without-juggler/repositories/account/datasources/mysqlconn.ts (hunk 25 above)
Comment (superkhau) [inline services/account-without-juggler/repositories/account/datasources/mysqlconn.ts]: move the require up to the top of the file

File: services/account-without-juggler/repositories/account/datasources/mysqlconn.ts [hunk 30]
```diff
@@ -0,0 +1,201 @@
...
+    return this.connection.connect();
+  }
+  disconnect() {
+    return this.connection.end();
+  }
+  ping() {
+    return this.connection.ping();
+  }
+
+  public updateAll(
```
Comment (superkhau) [inline services/account-without-juggler/repositories/account/datasources/mysqlconn.ts]: public is default, remove for consistency

File: services/account-without-juggler/repositories/account/datasources/mysqlconn.ts (hunk 25 above)
Comment (ssh24) [inline services/account-without-juggler/repositories/account/datasources/mysqlconn.ts]: That is not what our module (loopback-connector-mysql) uses though. Should we not test this with the same driver that our module currently uses?

File: services/account-without-juggler/repositories/account/datasources/mysqlconn.ts (hunk 29 above)
Comment (b-admike) [inline services/account-without-juggler/repositories/account/datasources/mysqlconn.ts]: We don't have that exported in Repository, but I was trying to make it of type `IConnection` from https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/mysql/index.d.ts#L39-#L83, but the return types for `connect` and `disconnect` don't align with ours (`Promise<void>`)

**Comment by b-admike** [comment]
@ssh24 @superkhau feedback applied, PTAL

File: services/account-without-juggler/repositories/account/datasources/mysqlconn.ts (hunk 29 above)
Comment (superkhau) [inline services/account-without-juggler/repositories/account/datasources/mysqlconn.ts]: Let's not prefix with I. You can create your own type or interface: See https://github.com/strongloop/loopback-next/blob/master/packages/core/src/internal-types.ts.  If we need to create a proxy to normalize the interfaces, so be it. cc @raymondfeng

File: services/account-without-juggler/controllers/AccountController.api.ts [hunk 31]
```diff
@@ -0,0 +1,114 @@
...
+    '/accounts/delete': {
+      delete: {
+        'x-operation-name': 'deleteById',
+        parameters: [
+          {
+            name: 'id',
+            in: 'query',
+            description: 'The id for the model instance to be deleted.',
+            required: true,
+             type:'object',
```
Comment (superkhau) [inline services/account-without-juggler/controllers/AccountController.api.ts]: nit, indent off by one space here
and id -> ID or identifier

File: services/account-without-juggler/controllers/AccountController.api.ts [hunk 32]
```diff
@@ -0,0 +1,114 @@
...
+        ],
+        responses: {
+          200: {
+            schema: {
+              type: 'object',
+              description: 'delete information',
+              properties: {
+                count: {
+                  type: 'number',
+                  description: 'number of records deleted',
```
Comment (superkhau) [inline services/account-without-juggler/controllers/AccountController.api.ts]: The number

File: services/account-without-juggler/controllers/AccountController.api.ts [hunk 33]
```diff
@@ -0,0 +1,114 @@
...
+        ],
+        responses: {
+          200: {
+            schema: {
+              type: 'object',
+              description: 'update information',
+              properties: {
+                count: {
+                  type: 'number',
+                  description: 'number of records updated',
```
Comment (superkhau) [inline services/account-without-juggler/controllers/AccountController.api.ts]: The number
and period at the end of your sentence

File: services/account-without-juggler/controllers/AccountController.api.ts [hunk 34]
```diff
@@ -0,0 +1,114 @@
...
+            name: 'id',
+            in: 'query',
+            description: 'The id of the model instance to be updated',
+            required: true,
+            type: 'string',
+          },
+         {
+            name: 'data',
+            in: 'body',
+            description: 'An object of model property name/value pairs',
```
Comment (superkhau) [inline services/account-without-juggler/controllers/AccountController.api.ts]: Period at the end for your description

File: services/account-without-juggler/controllers/AccountController.api.ts [hunk 35]
```diff
@@ -0,0 +1,114 @@
...
+      },
+    },
+    '/accounts/update': {
+      post: {
+        'x-operation-name': 'updateById',
+        parameters: [
+          {
+            name: 'id',
+            in: 'query',
+            description: 'The id of the model instance to be updated',
```
Comment (superkhau) [inline services/account-without-juggler/controllers/AccountController.api.ts]: Period at the end for your description.

File: services/account-without-juggler/controllers/AccountController.api.ts [hunk 36]
```diff
@@ -0,0 +1,114 @@
...
+            description: 'An object of model property name/value pairs',
+            required: true,
+            type: 'object',
+          },
+        ],
+        responses: {
+          200: {
+            schema: {
+              type: 'object',
+              description: 'update information',
```
Comment (superkhau) [inline services/account-without-juggler/controllers/AccountController.api.ts]: Full sentence description here pls.

File: services/account-without-juggler/controllers/AccountController.api.ts [hunk 37]
```diff
@@ -0,0 +1,114 @@
...
+            description: 'The id for the model instance to be deleted.',
+            required: true,
+             type:'object',
+          },
+        ],
+        responses: {
+          200: {
+            schema: {
+              type: 'object',
+              description: 'delete information',
```
Comment (superkhau) [inline services/account-without-juggler/controllers/AccountController.api.ts]: better desc here too

File: services/account-without-juggler/controllers/AccountController.ts [hunk 38]
```diff
@@ -0,0 +1,30 @@
+import { api } from 'loopback-next/packages/core';
+import { def } from './AccountController.api';
+import { accountRepository } from '../repositories/account';
+import { inject } from 'loopback-next/packages/context';
+
+@api(def)
+export class AccountController {
+  constructor(
+    @inject('repositories.account') private repository: accountRepository
```
Comment (superkhau) [inline services/account-without-juggler/controllers/AccountController.ts]: Shouldn't type be AccountRepository?

File: services/account-without-juggler/repositories/account/index.ts [hunk 39]
```diff
@@ -0,0 +1,12 @@
+import { CrudRepositoryImpl } from 'loopback-next/packages/repository';
+import { MySQLDs } from './datasources/mysqlds';
+import { Account } from './models/Account';
+
+
+export class accountRepository extends CrudRepositoryImpl<Account, string> {
```
Comment (superkhau) [inline services/account-without-juggler/repositories/account/index.ts]: Should be AccountRepository

File: services/account-without-juggler/tslint.json [hunk 40]
```diff
@@ -0,0 +1,72 @@
...
+    "variable-name": [true, "allow-leading-underscore", "ban-keywords", "check-format"],
+    "whitespace": [
+      true,
+      "check-branch",
+      "check-decl",
+      "check-separator",
+      "check-type",
+      "check-typecast",
+      "check-preblock"
+    ]
```
Comment (superkhau) [inline services/account-without-juggler/tslint.json]: Can you separate out this linting stuff into another PR?

**Comment by b-admike** [comment]
@superkhau PTAL

File: services/account-without-juggler/index.ts (hunk 14 above)
Comment (superkhau) [inline services/account-without-juggler/index.ts]: @b-admike Please link to the follow up issue here for posterity.

File: services/account-without-juggler/repositories/account/datasources/mysql.json [hunk 41]
```diff
@@ -0,0 +1,7 @@
+{
+    "host":"localhost",
+    "user":"root",
+    "password": "password",
+    "database": "account-service",
```
Comment (superkhau) [inline services/account-without-juggler/repositories/account/datasources/mysql.json]: Use the name of the example to distinguish from other databases the user may have locally before.

File: services/account-without-juggler/repositories/account/datasources/mysqlconn.ts [hunk 42]
```diff
@@ -0,0 +1,202 @@
+const debug = require('debug')('loopback:repositories:account:datasources:connections:mysql');
+const mysql = require('mysql');
+const db = require('mysql-promise')();
+import { CrudConnector } from 'loopback-next/packages/repository/src/crud-connector';
+import { IError, IConnection } from '@types/mysql';
```
Comment (superkhau) [inline services/account-without-juggler/repositories/account/datasources/mysqlconn.ts]: Do not use I for interfaces, should just be Error + Connection

File: services/account-without-juggler/repositories/account/datasources/mysqlds.ts [hunk 43]
```diff
@@ -0,0 +1,13 @@
...
+export class MySQLDs implements DataSource {
+  name: "mysqlDs";
+  connector: MySqlConn;
+  settings: Object;
+
+  constructor() {
+    this.settings = require("./mysql.json");
+    this.connector = new MySqlConn(this.settings);
+  }
+}
```
Comment (superkhau) [inline services/account-without-juggler/repositories/account/datasources/mysqlds.ts]: single quotes in the whole file pls

File: services/account-without-juggler/repositories/account/datasources/mysqlds.ts [hunk 44]
```diff
@@ -0,0 +1,13 @@
+import { MySqlConn } from "./mysqlconn";
+import { DataSource } from "loopback-next/packages/repository";
+
+export class MySQLDs implements DataSource {
+  name: "mysqlDs";
+  connector: MySqlConn;
+  settings: Object;
+
+  constructor() {
+    this.settings = require("./mysql.json");
```
Comment (superkhau) [inline services/account-without-juggler/repositories/account/datasources/mysqlds.ts]: move to top of file

File: services/account-without-juggler/repositories/account/index.ts [hunk 45]
```diff
@@ -0,0 +1,12 @@
+import { CrudRepositoryImpl } from 'loopback-next/packages/repository';
+import { MySQLDs } from './datasources/mysqlds';
+import { Account } from './models/Account';
+
+
+export class AccountRepository extends CrudRepositoryImpl<Account, string> {
+  constructor() {
+    const ds = new MySQLDs();
```
Comment (superkhau) [inline services/account-without-juggler/repositories/account/index.ts]: MySqlDs

File: services/account-without-juggler/repositories/account/models/Account.ts [hunk 46]
```diff
@@ -0,0 +1,12 @@
...
+  Entity,
+  ModelDefinition,
+  PropertyDefinition
+} from "loopback-next/packages/repository";
+
+@model(require("./account/model-definition"))
+export class Account extends Entity {
+  static definition = new ModelDefinition("Account", require("./account/model-definition").properties);
+  static modelName = "Account";
+}
```
Comment (superkhau) [inline services/account-without-juggler/repositories/account/models/Account.ts]: ditto for single quotes

File: services/account-without-juggler/repositories/account/models/account/model-definition.js (hunk 8 above)
Comment (superkhau) [inline services/account-without-juggler/repositories/account/models/account/model-definition.js]: single quotes and keys don't need quotes

File: services/account-without-juggler/test/unit/account-controller.ts [hunk 47]
```diff
@@ -0,0 +1,73 @@
...
+    expect(result.count).to.be.equal(1);
+    const getResult = await accCtrl.getAccount('{'where':{'id':'test1'}}');
+    expect(getResult).to.be.empty();
+  });
+});
+
+function createAccountController() {
+  accCtrl = new AccountController();
+  accCtrl.repository = new accountRepository();
+}
```
Comment (superkhau) [inline services/account-without-juggler/test/unit/account-controller.ts]: No abbreviations with var names pls

**Comment by superkhau** [comment]
Please update the README with working instructions too

File: services/account-without-juggler/index.ts (hunk 14 above)
Comment (b-admike) [inline services/account-without-juggler/index.ts]: Related issue: https://github.com/strongloop/loopback-next/issues/425

**Comment by b-admike** [comment]
@slnode test please
